ts/Series/CircularDataLabels.ts

```diff
@@ -0,0 +1,252 @@
+/* *
+ *
+ *  (c) 2010-2021 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import type SunburstSeries from './Sunburst/SunburstSeries.js';
+import type SunburstDataLabelOptions from '../Series/Sunburst/SunburstDataLabelOptions';
+import type DependencyWheelSeries from './DependencyWheel/DependencyWheelSeries.js';
+
+import U from '../Core/Utilities.js';
+const {
+    defined,
+    isNumber,
+    isObject,
+    merge,
+    splat
+} = U;
+
+/* *
+ *
+ *  Composition
+ *
+ * */
+
+namespace CircularDataLabels {
+
+    /* *
+     *
+     *  Declarations
+     *
+     * */
+    export interface CicrularDlOptionsParams extends SunburstSeries.DlOptionsParams {
+
+    }
+
+    export interface CicrularDlOptions extends SunburstDataLabelOptions {
+
+    }
+
+    /* *
+     *
+     *  Constants
+     *
+     * */
+
+    const rad2deg = 180 / Math.PI;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /* eslint-disable valid-jsdoc */
+
+    /** @private */
+    export function compose(
+        circularSeries: typeof SunburstSeries|DependencyWheelSeries
+    ): void {
+
+        (circularSeries as any).prototype.getDlOptions = getDlOptions;
+    }
+
+    function getDlOptions(
+        params: CicrularDlOptionsParams
+    ): any {
+        // Set options to new object to avoid problems with scope
+        let point = params.point,
+            shape: Partial<SunburstSeries.NodeValuesObject> =
+                isObject(params.shapeArgs) ? params.shapeArgs : {},
```

Comment:
Is it possible that `shapeArgs` is not an object? There's probably something wrong with TS